js/main.js

```diff
@@ -1,20 +1,68 @@
-import AddNewBook from './addNewBook.js';
-import removeBook from './removeBook.js';
-import showBooks from './showbooks.js';
-
-let collectionOfBooks = JSON.parse(localStorage.getItem('collection')) || [];
-
-document.addEventListener('DOMContentLoaded', () => {
-  showBooks(collectionOfBooks);
-});
-
-document.querySelector('.bookListCont').addEventListener('click', (e) => {
-  collectionOfBooks = removeBook(e, collectionOfBooks);
-  showBooks(collectionOfBooks);
-});
-
-document.getElementById('form').addEventListener('submit', (e) => {
-  e.preventDefault();
-  AddNewBook(e, collectionOfBooks);
-  showBooks(collectionOfBooks);
-});
+class MainBooks {
+  constructor() {
+    // ------------------------------------------ VARIABLES
+    this.collection = [];
+    this.title = document.getElementById('title');
+    this.author = document.getElementById('author');
+    this.submit = document.getElementById('form');
+    this.container = document.querySelector('.bookListCont');
+
+    this.displayBooks();
+
+    // ------------------------------------------- EVENTS
+    this.submit.addEventListener('submit', (e) => {
+      e.preventDefault();
+      // alert(123)
+      this.addBook();
+    });
+
+    this.container.addEventListener('click', (e) => {
+      if (e.target.classList.contains('remove')) {
+        const index = e.target.dataset.id;
+        this.removeBook(index);
+      }
+    });
+  }
+
+  // ------------------------ FUNCTIONS
+
+  addBook() {
+    const title = this.title.value;
+    const author = this.author.value;
+    this.collection.push({ title, author });
```

Comment:
- [ ] [OPTIONAl] It is better and more preferable to use the array methods that don't mutate the array, e.g. [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push), [pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) and so on.